packages/auth/src/core/auth/auth_impl.ts

```diff
@@ -325,19 +327,38 @@ export class AuthImpl implements AuthInternal, _FirebaseService {
       );
     }
 
+    if (!skipBeforeStateCallbacks) {
+      await this._runBeforeStateCallbacks(user);
+    }
+
     return this.queue(async () => {
       await this.directlySetCurrentUser(user as UserInternal | null);
       this.notifyAuthListeners();
     });
   }
 
+  async _runBeforeStateCallbacks(user: User | null): Promise<void> {
+    try {
```

Comment:
This method should probably check first whether or not the user is actually changing (like `directlySetCurrentUser` does).

Right now with this code, if you call auth.signOut() twice in a row, it will call the middleware twice in a row

Reply:
Added a check.